nearcore/src/lib.rs

```diff
@@ -74,17 +71,89 @@ pub fn get_default_home() -> PathBuf {
     PathBuf::default()
 }
 
+/// Returns the path of a directory where DB checkpoint can be found or created.
+/// Default location is the same as the database location: `path`.
+fn db_checkpoint_root_dir<'a>(path: &'a Path, near_config: &'a NearConfig) -> &'a Path {
+    if let Some(db_checkpoints_path) = &near_config.config.db_checkpoints_path {
+        &db_checkpoints_path
+    } else {
+        path
+    }
+}
+
+/// Returns the path of the DB checkpoint.
+fn db_checkpoint_path<'a>(path: &'a Path, near_config: &'a NearConfig) -> PathBuf {
+    let checkpoint_path = db_checkpoint_root_dir(path, near_config);
+    let checkpoint_path = checkpoint_path.join(DB_CHECKPOINT_NAME);
+    checkpoint_path
+}
```

Comment:
This feels a bit more verbose than it needs to be, something like this should work:

```suggestion
/// Returns the path of the DB checkpoint.
fn db_checkpoint_path<'a>(path: &'a Path, near_config: &'a NearConfig) -> PathBuf {
    near_config.config.db_checkpoints_path.unwrap_or(path).join(DB_CHECKPOINT_NAME)
}
```